lib/spack/spack/util/windows_registry.py

```diff
@@ -280,67 +328,51 @@ def _traverse_subkeys(self, stop_condition, collect_all_matching=False):
-        in which case call matches are returned.
-
-        Args:
-            search_key (str): regex string represeting a subkey name structure
-                              to be matched against.
-                              Cannot be provided alongside `direct_subkey`
-            collect_all_matching (bool): No-op if `direct_subkey` is specified
-        Return:
-            the desired subkey as a RegistryKey object, or none
-        """
-        return self._traverse_subkeys(search_key, collect_all_matching=collect_all_matching)
-
-    def find_subkey(self, subkey_name):
+    def find_subkey(self, subkey_name, recursive=True):
         """Perform a BFS of subkeys until desired key is found
         Returns None or RegistryKey object corresponding to requested key name
 
         Args:
-            subkey_name (str)
+            subkey_name (str): subkey to be searched for
+            recursive (bool):  perform a recursive search
         Return:
             the desired subkey as a RegistryKey object, or none
-
-        For more details, see the WindowsRegistryView._find_subkey_s method docstring
         """
-        return self._find_subkey_s(
-            WindowsRegistryView.KeyMatchConditions.name_matcher(subkey_name)
+        return self._traverse_subkeys(
+            WindowsRegistryView.KeyMatchConditions.name_matcher(subkey_name),
+            depth_search=recursive,
         )
 
-    def find_matching_subkey(self, subkey_name):
+    def find_matching_subkey(self, subkey_name, recursive=True):
         """Perform a BFS of subkeys until a key matching subkey name regex is found
         Returns None or the first RegistryKey object corresponding to requested key name
 
         Args:
-            subkey_name (str)
+            subkey_name (str): subkey to be searched for
+            recursive (bool):  perform a recursive search
         Return:
             the desired subkey as a RegistryKey object, or none
-
-        For more details, see the WindowsRegistryView._find_subkey_s method docstring
         """
-        return self._find_subkey_s(
-            WindowsRegistryView.KeyMatchConditions.regex_matcher(subkey_name)
+        return self._traverse_subkeys(
+            WindowsRegistryView.KeyMatchConditions.regex_matcher(subkey_name),
+            depth_search=recursive,
         )
 
-    def find_subkeys(self, subkey_name):
+    def find_subkeys(self, subkey_name, recursive=True):
         """Exactly the same as find_subkey, except this function tries to match
         a regex to multiple keys
 
         Args:
             subkey_name (str)
         Return:
             the desired subkeys as a list of RegistryKey object, or none
-
-        For more details, see the WindowsRegistryView._find_subkey_s method docstring
         """
-        kwargs = {"collect_all_matching": True}
-        return self._find_subkey_s(
+        kwargs = {"collect_all_matching": True, "depth_search": recursive}
```

Comment:
`_traverse_subkeys` now takes `recursive` vs. `depth_search`

Same issue in `find_matching_subkey` and `find_subkey`

Reply:
🤦